### PR TITLE
[TechDraw] Centralize centerat and paint functions

### PR DESCRIPTION
**Pull request by benj5378**
- This needs user testing
- This needs compiler testing
- Notice that not all paint functions have been centralized
- No one will believe how long it took to figure this code structure out

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [x]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [x]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [x]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

File: src/Mod/TechDraw/Gui/QGCustom.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+#ifndef _PreComp_
```
Comment (benj5378) [inline src/Mod/TechDraw/Gui/QGCustom.cpp:25]: What does this mean?

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7307">pipeline status</a> for feature branch [PR_7307](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7307). Pipeline [604286704](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/604286704) was triggered at [6e3db65](https://github.com/FreeCAD/FreeCAD/pull/7307/commits/6e3db654bfd17edc3a98448265a6ae3c02c1ce26). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by WandererFan** [comment]
Honestly, I don't know for sure.  I think it used to control if the
PreCompiled.h file was to be used, but the #def is commented out in
FCConfig.h.  It might be obsolete.

**Comment by luzpaz** [comment]
Experimental snap available of this PR: 

A build from commit 2032b9f7a7badca4ab4d42932e7cfbe2c69ff0df has been published to channel edge/pr-47 as revision 189.

  ## Installation instructions

  ```bash
    # Note: not all distros require 'sudo'
    $ sudo snap install --channel=edge/pr-47 freecad

    # If you already have the freecad snap installed
    $ sudo snap refresh --channel=edge/pr-47 freecad 

    # Experimental: Parallel install (see https://github.com/FreeCAD/FreeCAD-snap/blob/master/docs/index.md#parallel-installs)
    $ sudo snap set system experimental.parallel-instances=true
    $ sudo snap install --channel=edge/pr-47 freecad_pr47
    $ freecad_pr47 # Run it

    # To switch back to preferred snap (swap out 'edge' for 'stable' if necessary)
    $ snap refresh freecad --channel=edge freecad
  ```

**Comment by donovaly** [comment]
@benj5378, can you please rebase your PR?

@WandererFan, what is the status of this? Will you review it or is testing from others necessary?

**Comment by benj5378** [comment]
> @donovaly back in august-september I pinged @WandererFan multiple times about my pending PRs. First time, he said he would look at it, later, there was no reply.

Then I october/november, most of my PRs began getting closed without any warning or explantion. So I guess there isn't any interest for further of my PRs.

**Comment by WandererFan** [comment]
> @benj5378, can you please rebase your PR?
> 
> @WandererFan, what is the status of this? Will you review it or is testing from others necessary?

Still on my list to look at, but there is zero urgency to this one as it doesn't change any functionality.

**Comment by luzpaz** [comment]
Does this need to be simplified in to several PRs ?

**Comment by WandererFan** [comment]
<<Does this need to be simplified in to several PRs ?>>

No, it just needs to float up near the top of the TODO list.  It doesn't affect functionality and is only a minor maintainability improvement.

It's not going to merge nicely after all this time.  Could cancel it and replace with an issue I guess.

**Comment by luzpaz** [comment]
Closing in favor of FR #8227

**Pull request closed without merging**